ui/js/keypad.js

```diff
@@ -25,9 +25,10 @@ var Keypad = function (keypadId, opts, callback) {
 
     if (target.hasClass('enter')) {
       self.deactivate()
-      var result = self.type === 'phoneNumber'
-        ? libphonenumber.parsePhoneNumberFromString(self.result, self.opts.country).number
-        : self.result
+      var result = self.result
+      if (self.type === 'phoneNumber' && libphonenumber.parsePhoneNumberFromString(self.result, self.opts.country)) {
```

Comment:
We should only run parsePhoneNumberFromString once, and then check the variable to make sure it worked. Also, can result be a const?